editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -5929,12 +5931,25 @@ bool CanvasItemEditorViewport::can_drop_data(const Point2 &p_point, const Varian
 				if (!instantiated_scene) {
 					continue;
 				}
+
+				Node *edited_scene = EditorNode::get_singleton()->get_edited_scene();
+				if (_cyclical_dependency_exists(edited_scene->get_scene_file_path(), instantiated_scene)) {
+					memdelete(instantiated_scene);
+					can_instantiate = false;
+					is_cyclical_dep = true;
+					error_file = files[i];
+					break;
+				}
 				memdelete(instantiated_scene);
 			}
 			can_instantiate = true;
-			break;
 		}
 	}
+	if (is_cyclical_dep) {
+		canvas_item_editor->message = vformat(TTR("Can't instantiate: %s."), vformat(TTR("Circular dependency found at %s"), error_file.get_file().get_basename()));
```

Comment:
```suggestion
		canvas_item_editor->message = vformat(TTR("Can't instantiate: %s."), vformat(TTR("Circular dependency found at %s"), error_file.get_file()));
```
Should keep the extension IMO

Reply:
Only reason I removed the extension is to keep it similar to how it is done when `_perform_drop_data` does it:

https://github.com/godotengine/godot/blob/9b94c80e9aff2a4f363ae6d8e2bbe837aa5876bc/editor/plugins/canvas_item_editor_plugin.cpp#L5898

Reply:
Sure

Reply:
@KoBeWi thoughts on this? anyone else that might have feedback on this?

Reply:
tbh that other message could be tweaked. It doesn't even end with a period.
I think it's more clear if you include the extension.

Reply:
Sounds good, I'll include the extension.

That other message should be changed in a separate PR, right? Specially to find any other inconsistencies

Reply:
It's a simple fix, you could include it here. But either way it's fine.

Reply:
Whoops, I just force pushed without that change.

Reply:
@KoBeWi I'm retaking this along with the other change. What do you think about changing this:

https://github.com/godotengine/godot/blob/9b94c80e9aff2a4f363ae6d8e2bbe837aa5876bc/editor/plugins/canvas_item_editor_plugin.cpp#L5898

Which outputs (the last comma "," gets omitted somehow):

![image](https://github.com/godotengine/godot/assets/55825613/6d2f44e3-aade-4a9e-bf36-be79972504f7)

I wanted to add a space between each value: "file1.ext, file2.ext" but that would require a condition:

```
files_str += error_files[i].get_file();
if (i < error_files.size() - 1) {
 	files_str += ", ";
}
```

Which produces:

![image](https://github.com/godotengine/godot/assets/55825613/8f0dca13-678f-47ea-a3af-e6346f66b978)

Reply:
You could use PackedStringArray to store problematic items and then use `String(", ").join(file_array)` instead of `file_str`.

Reply:
That's pretty cool, didn't know about that! will check